mathics/core/evaluation.py

```diff
@@ -116,6 +116,11 @@ def run_with_timeout_and_stack(request, timeout):
 
     thread.join(timeout)
     if thread.is_alive():
```

Comment:
What is going on here is not obvious: that we are doing setting these variables because the behavior of Mathics interpretation looks for those specifically at certain points and uses them to abort execution. And the part about how this fails for long-running sympy (or numpy, etc) operations. 

And the link to the place that describes why a simple `join(timeout)` is not sufficient. 

I'd be happy to make a stab at this if you'd like.